### PR TITLE
Add polkadot tests

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

File: test/python/apps/polkadot.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    return bytes.fromhex(ss58_decode(address))
+
+def _format_amount(amount: int) -> bytes:
+    RuntimeConfiguration().update_type_registry(load_type_registry_preset("legacy"))
+    obj = RuntimeConfiguration().create_scale_object('Compact<Balance>')
+    scale_data = obj.encode(amount)
+    return bytes(scale_data.get_remaining_bytes())
+
+# Not sure what this exactly is but we don't actually care
+UNKNOWN = bytes([0x85, 0x02, 0x00, 0x00])
```
Comment (agrojean-ledger) [inline test/python/apps/polkadot.py:36]: Can't we just put a name on what this is ? ^^

File: test/python/apps/polkadot.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+               + _format_amount(send_amount) \
+               + UNKNOWN \
+               + SPEC_VERSION.to_bytes(4, "little") \
+               + TX_VERSION.to_bytes(4, "little") \
+               + GENESIS_HASH \
+               + BLOCK_HASH
+
+    def craft_invalid_polkadot_transaction(address, send_amount, fees, memo) -> bytes:
+        force_transfer = Method.BALANCE_FORCE_TRANSFER.to_bytes(2, "big") \
+                       + bytes([0x00, 0xdc, 0x5a, 0xda, 0x10, 0xee, 0xdd, 0x89, 0x81, 0x92,
```
Comment (agrojean-ledger) [inline test/python/apps/polkadot.py:138]: Could we reuse this invalid byte array in other tests ? If so, maybe put it in a variable

File: test/python/test_polkadot.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        # Get public key.
+        key = dot.get_pubkey()
+        # Init signature process and assert response APDU code is 0x9000 (OK).
+        dot.sign_init().status
+        # craft tx
+        message = PolkadotClient.craft_valid_polkadot_transaction(destination, send_amount, None, None)
+        # Send message to be signed
+        sign_response = dot.sign_last(message)
+
+        # Assert signature is verified properly with key and message
```
Comment (agrojean-ledger) [inline test/python/test_polkadot.py:38]: You have remove the assert on the return code ?

File: test/python/apps/polkadot.py (hunk 1 above)
Comment (fbeutin-ledger) [reply test/python/apps/polkadot.py:36]: I would like to, but I spent a lot of time trying to figure out what this does to no avail... Not sure we would gain a lot

File: test/python/apps/polkadot.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+               + AccountIdLookupType.ID.to_bytes(1, "big") \
+               + _polkadot_address_to_pk(address) \
+               + _format_amount(send_amount) \
+               + UNKNOWN \
+               + SPEC_VERSION.to_bytes(4, "little") \
+               + TX_VERSION.to_bytes(4, "little") \
+               + GENESIS_HASH \
+               + BLOCK_HASH
+
+    def craft_invalid_polkadot_transaction(address, send_amount, fees, memo) -> bytes:
```
[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/polkadot.py:136]: mmmmmh actually true, I planned to add them for everyone but ended up not doing it ...

[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/polkadot.py:136]: I'll see if I can quickly draft something

[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/polkadot.py:136]: Ok I have a working prototype but only Polkadot implements the `craft_invalid_polkadot_transaction` callback for now so I can't merge without implementing it for everyone
I suggest doing it in a dedicated branch and merging it later

File: test/python/apps/polkadot.py (hunk 2 above)
Comment (fbeutin-ledger) [reply test/python/apps/polkadot.py:138]: This function will be called by every test doing a `wrong method`

File: test/python/apps/polkadot.py (hunk 1 above)
Comment (agrojean-ledger) [reply test/python/apps/polkadot.py:36]: Ok, no worries, let's go ahead with unknown.

File: test/python/apps/polkadot.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (agrojean-ledger) [reply test/python/apps/polkadot.py:136]: Can we have an issue or a PR ?

File: test/python/test_polkadot.py (hunk 3 above)
Comment (agrojean-ledger) [reply test/python/test_polkadot.py:38]: Ok.

File: test/python/apps/polkadot.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/polkadot.py:136]: https://github.com/LedgerHQ/app-exchange/pull/150

**Review by agrojean-ledger — APPROVED** [review]

**Pull request merged**